src/query/graphite/graphite/timespec.go

```diff
@@ -108,9 +137,146 @@ func ParseTime(s string, now time.Time, absoluteOffset time.Duration) (time.Time
 		return time.Unix(n, 0).UTC(), nil
 	}
 
+	ref, offset := s, ""
+	if strings.Contains(s, "+") {
+		stringSlice := strings.Split(s, "+")
+		ref, offset = stringSlice[0], stringSlice[1]
+		offset = "+" + offset
+	} else if strings.Contains(s, "-") {
+		stringSlice := strings.Split(s, "-")
+		ref, offset = stringSlice[0], stringSlice[1]
+		offset = "-" + offset
+	}
+	err = nil
+	parsedReference, err := ParseTimeReference(ref, now)
```

Comment:
Need to check err here before proceeding to next line?

Maybe this could be rewritten:
```go
if parsedReference, err := ParseTimeReference(ref, now); err == nil {
  if parsedOffset, err := ParseOffset(offset); err == nil {
    return parsedReference.Add(parsedOffset), nil
  }
}
```

Reply:
great idea